### ISSUE TITLE
[BUG] ProxyToolbar.logo: expected a value of type str, got None

### ISSUE THREAD

**Issue by kernc** [bokeh/bokeh#10880]
#### ALL software version info (bokeh, python, notebook, OS, browser, any other relevant packages)

Regression in bokeh-2.3.0.dev12
Works in bokeh-2.3.0.dev11

#### Description of expected behavior and the observed behavior

Passing `gridplot(..., toolbar_options=dict(logo=None))` should hide the toolbar logo.
Instead, since bokeh-2.3.0.dev12, it raises the error below.

#### Complete, minimal, self-contained example code that reproduces the issue

```py
from bokeh.io import show
from bokeh.layouts import gridplot
from bokeh.plotting import figure

p = figure()
p.line([1, 2, 3, 4], [1, 4, 3, 0])
fig = gridplot([[p]], toolbar_options=dict(logo=None))
show(fig)
```

#### Stack traceback and/or browser JavaScript console output
```
Traceback (most recent call last):
  File "test.py", line 7, in <module>
    fig = gridplot([[p]], toolbar_options=dict(logo=None))
  File "~/venv/lib/python3.7/site-packages/bokeh/layouts.py", line 306, in gridplot
    proxy = ProxyToolbar(toolbars=toolbars, tools=tools, **toolbar_options)
  File "~/venv/lib/python3.7/site-packages/bokeh/model.py", line 236, in __init__
    super().__init__(**kwargs)
  File "~/venv/lib/python3.7/site-packages/bokeh/core/has_props.py", line 268, in __init__
    setattr(self, name, value)
  File "~/venv/lib/python3.7/site-packages/bokeh/core/has_props.py", line 297, in __setattr__
    super().__setattr__(name, value)
  File "~/venv/lib/python3.7/site-packages/bokeh/core/property/descriptors.py", line 551, in __set__
    self._internal_set(obj, value, setter=setter)
  File "~/venv/lib/python3.7/site-packages/bokeh/core/property/descriptors.py", line 783, in _internal_set
    value = self.property.prepare_value(obj, self.name, value)
  File "~/venv/lib/python3.7/site-packages/bokeh/core/property/bases.py", line 347, in prepare_value
    raise ValueError(f"failed to validate {obj_repr}.{name}: {error}")
ValueError: failed to validate ProxyToolbar(id='1049', ...).logo: expected a value of type str, got None of type NoneType
```

**Comment by kernc** [comment]
Without firing up bisect, I'd say it has to be https://github.com/bokeh/bokeh/commit/6828be458c924193fdbed49126b56358da5b365c from https://github.com/bokeh/bokeh/pull/10842.

It's not quite clear to me how to pass a Nullable type (passing `logo=bokeh.core.property.primitive.Null()` didn't work), but the docs still mention setting the value to `None`:
https://docs.bokeh.org/en/latest/docs/reference/models/tools.html?highlight=logo#bokeh.models.tools.Toolbar.logo

**Comment by mattpap** [comment]
We are still finding cases that implicitly allowed `None`, but it wasn't indicated in the type system. It's fixed in PR #10874.

**Comment by mattpap** [comment]
`Null` and `Nullable` are property types that accept `None` (`null` in bokehjs) as a value. You can use the former in `Either(Null, T0, ..., Tn)` and the later in `Nullable(Type)`.
